cc3d/CompuCellSetup/persistent_globals.py

```diff
@@ -96,18 +98,32 @@ def set_output_dir(self, output_dir: str) -> None:
 
         self.__output_dir = output_dir
 
+    def set_workspace_dir(self, workspace_dir: str) -> None:
+        """
+        Sets screenshot dir - usually this is a custom screenshot directory set based on
+        command line input
+
+        :param workspace_dir:
+        :return:
+        """
+
+        self.__workspace_dir = workspace_dir
+
     @property
     def workspace_dir(self) -> str:
         """
         returns workspace directory
         :return:
         """
 
-        workspace_dir = os.path.join(os.path.expanduser('~'), 'CC3DWorkspace')
-        if not exists(workspace_dir):
-            Path(workspace_dir).mkdir(parents=True, exist_ok=True)
+        if self.__workspace_dir is None:
+            workspace_dir = os.path.join(os.path.expanduser('~'), 'CC3DWorkspace')
+            if not exists(workspace_dir):
+                Path(workspace_dir).mkdir(parents=True, exist_ok=True)
 
-        return workspace_dir
+            return workspace_dir
+        else:
+            return self.__workspace_dir
```

Comment:
Wouldn't it be interesting and user friendly to make the user inputed directory if it doesn't exist?

Reply:
Agreed.

Reply:
It is already implemented